addon/utils/proxy-unproxy-objects.js

```diff
@@ -0,0 +1,46 @@
+import { isNone } from '@ember/utils';
+/**
+ * This utility is ussed to create a wrapper object around the object passed to the proxy function.
+ * Currently on drag action, the `draggable-object` mutates the `content` object and appends a new
+ * property `isDraggingObject` onto the parent object.
+ * This behavior throws an exception with M3 models because it not a previously whitelisted property. The issue
```

Comment:
let's avoid mentioning M3 and instead be a bit more abstract.

> This unexpected mutation causes problems when the targeted content is not prepared to handle the additional property, and potentially leaks local state onto an object that likely holds state for the route or application more generally.

Reply:
+1
Resolved